From 33bc81d5a262e0fa1a346ac1667a3679a17c0997 Mon Sep 17 00:00:00 2001
From: Ross Burton <ross.burton@intel.com>
Date: Tue, 3 Jul 2018 13:59:09 +0100
Subject: [PATCH] Make CPU family warnings fatal

Rebased for meson 1.12.0: detect_cpu_family() moved from mesonbuild/environment.py
into mesonbuild/envconfig.py, so both hunks now target the one file. The
EnvironmentException this raises is already imported there.

Also corrects "trail" to "trial" in the second hunk. The old patch tested a name
that does not exist, so on an unknown CPU family it would have raised NameError
instead of the intended EnvironmentException.

Upstream-Status: Inappropriate [OE specific]
Signed-off-by: Ross Burton <ross.burton@intel.com>
---
 mesonbuild/envconfig.py | 10 ++++------
 1 file changed, 4 insertions(+), 6 deletions(-)

diff --git a/mesonbuild/envconfig.py b/mesonbuild/envconfig.py
index 3569927..80a9535 100644
--- a/mesonbuild/envconfig.py
+++ b/mesonbuild/envconfig.py
@@ -295,8 +295,8 @@ class MachineInfo(HoldableObject):
                 'but is missing {}.'.format(minimum_literal - set(literal)))
 
         cpu_family = literal['cpu_family']
-        if cpu_family not in known_cpu_families:
-            mlog.warning(f'Unknown CPU family {cpu_family}, please report this at https://github.com/mesonbuild/meson/issues/new')
+        if cpu_family not in known_cpu_families and cpu_family != "riscv":
+            raise EnvironmentException(f'Unknown CPU family {cpu_family}, see https://wiki.yoctoproject.org/wiki/Meson/UnknownCPU for directions.')
 
         endian = literal['endian']
         if endian not in ('little', 'big'):
@@ -657,10 +657,8 @@ def detect_cpu_family(compilers: CompilerDict) -> str:
         if compilers and not any_compiler_has_define(compilers, '__mips64'):
             trial = 'mips'
 
-    if trial not in known_cpu_families:
-        mlog.warning(f'Unknown CPU family {trial!r}, please report this at '
-                     'https://github.com/mesonbuild/meson/issues/new with the '
-                     'output of `uname -a` and `cat /proc/cpuinfo`')
+    if trial not in known_cpu_families and trial != "riscv":
+        raise EnvironmentException(f'Unknown CPU family {trial!r}, see https://wiki.yoctoproject.org/wiki/Meson/UnknownCPU for directions.')
 
     return trial
 
-- 
2.34.1
